modules/custom/az_news/az_news_export/src/Plugin/views/row/AZNewsDataFieldRow.php

```diff
@@ -56,152 +86,201 @@ public static function create(ContainerInterface $container, array $configuratio
+      'entity_reference',
+      'entity_reference_revisions',
+      'file',
+    ];
+    $field_type = $field_definition->getType();
+    $field_target_type = $field_definition->getSetting('target_type');
+    return (in_array($field_type, $reference_field_types) && $field_target_type === $target_type);
+  }
+
+  /**
+   * Generalized method to serialize fields.
+   *
+   * @param mixed $value
+   *   The field value(s), can be an array of entity IDs or a single entity ID.
+   * @param string $target_type
+   *   The type of entities the field references
+   *   ('media', 'taxonomy_term', etc.).
+   *
+   * @return array
+   *   The serialized field value.
+   */
+  protected function serializeField($value, $target_type): array {
+    $items = [];
+    // Normalize $value to an array to simplify processing.
+    $values = is_array($value) ? $value : [$value];
+    if (!empty($values)) {
+      $storage = $this->entityTypeManager->getStorage($target_type);
+      $entities = $storage->loadMultiple($values);
+
+      foreach ($entities as $referencedEntity) {
+        if (!$referencedEntity->access('view')) {
+          continue;
         }
-        return $items;
-      },
-      // Serialize contacts as associative arrays of keyed fields.
-      'field_az_contacts' => function ($value, $entity) {
-        $items = [];
-        $contacts = $this->entityTypeManager->getStorage('paragraph')->loadMultiple($value);
-        foreach ($contacts as $contact) {
-          if (!$contact->access('view')) {
-            continue;
-          }
-          $item = [];
-          // Serialize specific fields from contact.
-          $contact_fields = [
-            'field_az_email',
-            'field_az_title',
-            'field_az_phone',
-          ];
-          foreach ($contact_fields as $contact_field) {
-            if ($contact->hasField($contact_field) && !empty($contact->{$contact_field}->value)) {
-              $item[$contact_field] = $contact->{$contact_field}->value;
+
+        $item = [];
+        switch ($target_type) {
+          case 'media':
+            if ($referencedEntity instanceof MediaInterface) {
+              $media_type = $referencedEntity->bundle();
+              $fid = $referencedEntity->getSource()->getSourceFieldValue($referencedEntity);
+              $file = $this->entityTypeManager->getStorage('file')->load($fid);
+              $item['fid'] = $file->id();
+              $item['uuid'] = $file->uuid();
+              switch ($media_type) {
+                case 'az_image':
+                  $item['original'] = $file->createFileUrl(FALSE);
+                  $uri = $file->getFileUri();
+                  $styles = [
+                    'thumbnail' => 'az_enterprise_thumbnail',
+                    'thumbnail_small' => 'az_enterprise_thumbnail_small',
+                  ];
+                  foreach ($styles as $key => $style_id) {
+                    $image_style = ImageStyle::load($style_id);
```

Comment:
Might as well use the injected Entity Type Manger service for this (?)
```suggestion
                    $image_style = $this->entityTypeManager->getStorage('image_style')->load($style_id);
```